physicsnemo/models/module.py

```diff
@@ -332,6 +400,42 @@ def save(self, file_name: Union[str, None] = None, verbose: bool = False) -> Non
             If file_name does not end with .mdlus extension
         """
 
+        # Define some helper functions
+        def _save_process(module, args, metadata, mod_prefix="") -> None:
+            """Helper function to recursively populate the dictionaries args
+            and metadata required for saving the module and its submodules."""
+
+            # Pointer to args["__args__"] for submodules
+            if mod_prefix == "":
+                args_ptr = args["__args__"].copy()
+            else:
+                args_ptr = args[mod_prefix]["__args__"].copy()
+
+            for arg_name, arg_value in args_ptr.items():
+                if isinstance(arg_value, Module):
+                    next_mod_prefix = (
+                        f"{mod_prefix if mod_prefix else _BASE_CKPT_PREFIX}.{arg_name}"
+                    )
+                    args[next_mod_prefix] = arg_value._args.copy()
+                    args_ptr[arg_name] = next_mod_prefix
+                    metadata[f"{next_mod_prefix}.mdlus_file_version"] = (
+                        arg_value.__model_checkpoint_version__
+                    )
+                    _save_process(arg_value, args, metadata, next_mod_prefix)
+                elif isinstance(arg_value, torch.nn.Module):
+                    warnings.warn(
```

Comment:
I would think instead of a warning you would actually want to raise an error if this is the case